### PR TITLE
SDIT-937 Add Search by BookingId tests

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/config/GsonConfig.kt [hunk 1]
```diff
@@ -17,40 +17,33 @@ import java.time.format.DateTimeFormatter
 import java.util.Locale
```
Comment (karenmillermoj) [inline hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/config/GsonConfig.kt:17]: Just a tidy up on this file

File: hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/resource/PrisonerSearchByBookingIdsResourceTest.kt [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  override fun loadPrisonerData() {
+    val prisonerNumbers = List(12) { i -> "AN$i" }
+    var prisoners = ArrayList<Prisoner>()
+    prisonerNumbers.forEachIndexed { bookingId: Int, prisonNumber: String ->
+      val offenderBooking = getOffenderBooking(prisonNumber, bookingId.toLong())
+      val prisoner = Prisoner(offenderBooking, null, null)
+      prisoners.add(prisoner)
+    }
+    loadPrisoners(prisoners)
+  }
```
Comment (karenmillermoj) [inline hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/resource/PrisonerSearchByBookingIdsResourceTest.kt:22]: This was the main change - to mimic how it was previously loading the data (via mocked out api calls)

File: hmpps-prisoner-search/src/test/resources/application-test.yml [hunk 3]
```diff
@@ -1,9 +1,6 @@
 server:
   shutdown: immediate
 
-aws:
```
Comment (karenmillermoj) [inline hmpps-prisoner-search/src/test/resources/application-test.yml:4]: Not needed

File: hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/config/GsonConfig.kt [hunk 4]
```diff
@@ -17,40 +17,33 @@ import java.time.format.DateTimeFormatter
...
-  fun gson(): Gson {
-    return GsonBuilder().setPrettyPrinting()
-      .registerTypeAdapter(LocalDate::class.java, LocalDateAdapter())
-      .registerTypeAdapter(LocalDateTime::class.java, LocalDateTimeAdapter())
-      .create()
-  }
+  fun gson(): Gson = GsonBuilder().setPrettyPrinting()
+    .registerTypeAdapter(LocalDate::class.java, LocalDateAdapter())
+    .registerTypeAdapter(LocalDateTime::class.java, LocalDateTimeAdapter())
+    .create()
```
Comment (mikehalmamoj) [inline hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/config/GsonConfig.kt:26]: I thnk Pete moved the indexer away from Gson to Jackson so maybe we should do the same at some point

**Review by mikehalmamoj — APPROVED** [review]

File: hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/config/GsonConfig.kt (hunk 4 above)
Comment (karenmillermoj) [reply hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/config/GsonConfig.kt:26]: yes - I agree. Cheers

**Pull request merged**